File: vz03/src/renders/habitats/water_habitat.cpp
/* NIM/Nama  : 13515AAA / AAA       */
/* File      : Habitat.cpp              */

#include "water_habitat.h"

using namespace std;

WaterHabitat::WaterHabitat() : Habitat('W', WHITE)
{
}
WaterHabitat::WaterHabitat(const WaterHabitat& W) : Habitat(W)
{
}
WaterHabitat::~WaterHabitat()
{
}
WaterHabitat& WaterHabitat::operator=(const WaterHabitat& W)
{
	Habitat::operator=(W);
	return *this;
}
char WaterHabitat::render()
{
	return id;
}
Color WaterHabitat::getColor()
{
	return color;
}
WaterHabitat* WaterHabitat::clone() const
{
	return new WaterHabitat(*this);
}

File: WaterHabitat.java
/* NIM/Nama : 13515078 / Veren Iliana Kurniadi */
/* File     : WaterHabitat.java                */

import java.util.*;
import java.lang.*;

class WaterHabitat extends Habitat{
   /**
   * Constructor
   */
  public WaterHabitat(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return 'W';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
